tests/bit_vector/find_next_set_test.cu

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <utils.hpp>
+
+#include <cuco/detail/trie/bit_vector/bit_vector.cuh>
+
+#include <thrust/device_vector.h>
+#include <thrust/execution_policy.h>
+#include <thrust/host_vector.h>
+
+#include <catch2/catch_test_macros.hpp>
+
+template <class BitVectorRef>
+__global__ void find_next_set_kernel(BitVectorRef ref, size_t n, uint64_t* output)
```

Comment:
If you add a tparam `ÒutputIt`, you won't need the `thrust::raw_pointer_cast` at the call site.

Reply:
6028a794dff6a1e3daf1a8b4d3980ee7c172601b